python/langsmith/schemas.py

```diff
@@ -1109,6 +1109,45 @@ class UsageMetadata(TypedDict):
 
     Does *not* need to sum to full output token count. Does *not* need to have all keys.
     """
+    input_cost: NotRequired[Decimal]
+    """The cost of the input tokens."""
+    output_cost: NotRequired[Decimal]
+    """The cost of the output tokens."""
+    total_cost: NotRequired[Decimal]
+    """The total cost of the tokens."""
+    input_cost_details: NotRequired[dict[str, Any]]
+    """The cost details of the input tokens."""
+    output_cost_details: NotRequired[dict[str, Any]]
+    """The cost details of the output tokens."""
+
+
+class ExtractedUsageMetadata(TypedDict, total=False):
```

Comment:
not sure i follow, doesn't UsageMetadata already not require all keys to be present? is the issue that we want to make input_tokens/output_tokens/etc also not required?